src/main/java/org/opensearch/sdk/SDKClient.java

```diff
@@ -460,12 +473,12 @@ public void delete(DeleteRequest request, ActionListener<DeleteResponse> listene
          * @see Requests#searchRequest(String...)
          */
         public void search(SearchRequest request, ActionListener<SearchResponse> listener) {
-            restHighLevelClient.searchAsync(request, RequestOptions.DEFAULT, listener);
+            RequestOptions options = RequestOptions.DEFAULT.toBuilder().addHeader(this.headerName, "Bearer" + this.token).build();
```

Comment:
Let's not construct the `options` here.  Have the user write this code and set the RequestOptions somewhere (with a default of `RequestOptions.DEFAULT` if it's not specified).

There are two ways of doing this.

1. Have an instance field `RequestOptions options` that you use here, and have a setter for it.
2. Add a new method that takes the options as the default.  (Probably not the best idea because we already have that ability to do with the client itself, which we are calling with `searchAsync()`.)

So I'd go with option 1.